src/osdc/QosProfileMgr.cc

```diff
@@ -0,0 +1,57 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2021 Red Hat Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#include <memory>
+#include <mutex>
+
+#include "QosProfileMgr.h"
+
+osdc::qos_profile_ref osdc::QosProfileMgr::create(uint64_t res,
+                                                  uint64_t wgt,
+                                                  uint64_t lim)
+{
+  return create(res, wgt, lim, next_client_profile_id++);
```

Comment:
next_client_profile_id isn't protected by the lock here -- you probably need to create a create_helper() so that the two create() overloads can take the lock and call it.

Reply:
Yes, that's correct! create_qos_profile() is now called by the two create() overloads after acquiring the lock.